lib/concurrent/executor/ruby_thread_pool_executor.rb

```diff
@@ -269,17 +287,20 @@ class Worker
 
       def initialize(pool)
         # instance variables accessed only under pool's lock so no need to sync here again
-        @queue  = Queue.new
         @pool   = pool
+        # I'm not sure that we need a prioritized queue here
+        # Queued jobs should be in the pool's queue, this is mainly for synchronization
+        #@queue  = pool.prioritized? ? Concurrent::PriorityBlockingQueue.new(order: :max) : Queue.new
+        @queue  = Queue.new
```

Comment:
@pitr-ch If my understanding is correct, we do not need a prioritized queue here, do we? This queue is mainly used to wake up an idle thread. If jobs are backed up in the queue they will be backed up in the pool's queue. Is this correct?